src/GraphQl/Subscription/MercureSubscriptionUriGeneratorInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\GraphQl\Subscription;
+
+/**
+ * Generates Mercure-related URIs from a subscription ID.
+ *
+ * @experimental
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+interface MercureSubscriptionUriGeneratorInterface
+{
+    public function generateMercureTopicIri(string $subscriptionId): string;
+
+    public function generateMercureTopicUrl(string $subscriptionId): string;
+
+    public function generateMercureUrl(string $subscriptionId): string;
```

Comment:
I suggest to simplify this interface and to provide only a single method `getTopic(string $subscriptionId)`

Reply:
I can remove the `generateMercureTopicIri` but I need two methods: one for getting the topic IRI and one for getting the Mercure URL (with the topic in the query).